### ISSUE TITLE
[HxGrid] How to add more than 1 custom column?

### ISSUE THREAD

**Issue by surgicalcoder** [havit/Havit.Blazor#522]
I'm trying to create more than 1 custom column on a Grid, the use case is to hold buttons or other controls,

`HxGridColumn` does not contain any `ChildContent`, only a `HeaderTemplate` and a `PlaceholderTemplate`

`HxContextMenuGridColumn<T>` seems to do the job, however you can only add one those on the grid, because of [this line here](https://github.com/havit/Havit.Blazor/blob/3f98fe00267daef2b2b2f0e5e5215238c260bd76/Havit.Blazor.Components.Web.Bootstrap/Grids/HxGrid.razor.cs#L547), and because it's effectively hard coded in [HxContextMenuGridColumn](https://github.com/havit/Havit.Blazor/blob/3f98fe00267daef2b2b2f0e5e5215238c260bd76/Havit.Blazor.Components.Web.Bootstrap/Grids/HxContextMenuGridColumn.cs#L45), you can't add more than one custom column,

Can this get changed to accept a parameter, so you can have more than one, or perhaps copy the same logic from [HxGridColumn](https://github.com/havit/Havit.Blazor/blob/3f98fe00267daef2b2b2f0e5e5215238c260bd76/Havit.Blazor.Components.Web.Bootstrap/Grids/HxGridColumn.cs#L15) ?

**Comment by pollardpj** [comment]
I may have misunderstood your issue, but I use:

`[Parameter] public RenderFragment<TItem> ItemTemplate { get; set; }`

 on HxGridColumn to include whatever I want in a column, buttons, custom html etc. 

Is this what you need?

**Comment by surgicalcoder** [comment]
@pollardpj Very possibly, genuinely not sure how I missed that 😆 

What does HxContextMenuGridColumn give you that HxGridColumn doesn't? Or is that the reason it might be removed in vNext?

In that case, maybe the documentation needs an example, I didn't find it obvious hence why I went down the Context Menu route...

**Comment by pollardpj** [comment]
Hehe, give it a go, if you still have issues, I may have faced them myself, so let me know here: I find that the Havit Library is very flexible, always has the solution to a problem I have. I had looked at `HxContextMenuGridColumn<TItem>` for my projects, which is useful, but settled on `ItemTemplate` as it gave me what I specifically wanted.

**Comment by hakenr** [comment]
You are right, the `HxContextMenuGridColumn` is a very simple preconfiguration of `HxGridColumnBase` and does not bring much value over `HxGridColumn` except for better readability.

(What we will probably remove is the `HxGrid.ContextMenu` parameter (`RenderFragment`) which was originally inteded to be used for context menus but turned out to be difficult to customize. Thus we decided for `HxContextMenuGridColumn` or `HxGridColumn` itself to be more flexible.)
